packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -168,3 +168,147 @@ bool _assertOutOfExtent(double extent) {
+
+  @override
+  void performLayout() {
+    double offset = 0;
+    double maxPaintExtent = 0;
+
+    RenderSliver? child = firstChild;
+
+
+    while (child != null) {
+      final double beforeOffsetPaintExtent = calculatePaintOffset(
+        constraints,
+        from: 0.0,
+        to: offset,
+      );
+      child.layout(
+        constraints.copyWith(
+          scrollOffset: math.max(0.0, constraints.scrollOffset - offset),
+          cacheOrigin: math.min(0.0, constraints.cacheOrigin + offset),
+          overlap: math.max(0.0, constraints.overlap - beforeOffsetPaintExtent),
+          remainingPaintExtent: constraints.remainingPaintExtent - beforeOffsetPaintExtent,
+          remainingCacheExtent: constraints.remainingCacheExtent - calculateCacheOffset(constraints, from: 0.0, to: offset),
+          precedingScrollExtent: offset + constraints.precedingScrollExtent,
+        ), parentUsesSize: true
+      );
+      final SliverGeometry childLayoutGeometry = child.geometry!;
+      final SliverPhysicalParentData childParentData = child.parentData! as SliverPhysicalParentData;
+      switch (constraints.axis) {
+        case Axis.vertical:
+          childParentData.paintOffset = Offset(0.0, beforeOffsetPaintExtent);
+        case Axis.horizontal:
+          childParentData.paintOffset = Offset(beforeOffsetPaintExtent, 0.0);
+      }
+      offset += childLayoutGeometry.scrollExtent;
+      maxPaintExtent += child.geometry!.maxPaintExtent;
+      child = childAfter(child);
+    }
+
+    final double totalScrollExtent = offset;
+    offset = 0.0;
+    child = firstChild;
+    // Second pass to correct out of bound paintOffsets.
+    while (child != null) {
+      final double beforeOffsetPaintExtent = calculatePaintOffset(
+        constraints,
+        from: 0.0,
+        to: offset,
+      );
+      final SliverGeometry childLayoutGeometry = child.geometry!;
+      final SliverPhysicalParentData childParentData = child.parentData! as SliverPhysicalParentData;
+      final double remainingExtent = totalScrollExtent - constraints.scrollOffset;
+      if (childLayoutGeometry.paintExtent > remainingExtent) {
+        final double paintCorrection = childLayoutGeometry.paintExtent - remainingExtent;
+        switch (constraints.axis) {
+          case Axis.vertical:
+            childParentData.paintOffset = Offset(0.0, beforeOffsetPaintExtent - paintCorrection);
+          case Axis.horizontal:
+            childParentData.paintOffset = Offset(beforeOffsetPaintExtent - paintCorrection, 0.0);
+        }
+      }
+      offset += child.geometry!.scrollExtent;
+      child = childAfter(child);
+    }
+    geometry = SliverGeometry(
+      scrollExtent: totalScrollExtent,
+      paintExtent: calculatePaintOffset(constraints, from: 0, to: totalScrollExtent),
+      maxPaintExtent: maxPaintExtent,
+    );
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    RenderSliver? child = lastChild;
```

Comment:
I think visitChildForSemantics expects the children to be visited in paint order, so that may need to be overridden to work with this reverse paint order.

Reply:
Gotcha. This is the paint order because this is how `Viewport` paints its sliver children. I think it's necessary for `SliverPersistentHeader` to be painted on top of the widgets that come after it.

Reply:
Are you going to implement visitChildrenForSemantics?